Packages/Networking/Sources/Networking/GraphQLNetworkSession.swift

```diff
@@ -0,0 +1,97 @@
+        let requestChainTransport = RequestChainNetworkTransport(interceptorProvider: provider,
+                                                                 endpointURL: url)
+
+        return ApolloClient(networkTransport: requestChainTransport,
+                            store: store)
+    }()
+
+    public func query<T: GraphQLQuery>(_ query: T) async throws -> T.Data {
+        try await withCheckedThrowingContinuation { continuation in
+            client.fetch(query: query) { result in
+                switch result {
+                case .success(let graphQLResult):
+                    if let data = graphQLResult.data {
+                        continuation.resume(returning: data)
+                    }
+                    if graphQLResult.errors != nil {
+                        continuation.resume(throwing: ApolloError())
+                    }
+                case .failure(let error):
+                    continuation.resume(throwing: error)
+                }
+            }
+        }
+    }
+
+    public func mutate<T: GraphQLMutation>(_ mutation: T) async throws -> T.Data {
+        try await withCheckedThrowingContinuation { continuation in
+            client.perform(mutation: mutation) { result in
+                switch result {
+                case .success(let graphQLResult):
+                    if let data = graphQLResult.data {
+                        continuation.resume(returning: data)
+                    }
+                    if graphQLResult.errors != nil {
+                        continuation.resume(throwing: ApolloError())
+                    }
+                case .failure(let error):
+                    continuation.resume(throwing: error)
+                }
+            }
+        }
+    }
+}
+
+class NetworkInterceptorProvider: DefaultInterceptorProvider {
+    override func interceptors<Operation: GraphQLOperation>(for operation: Operation) -> [ApolloInterceptor] {
+        var interceptors = super.interceptors(for: operation)
+        interceptors.insert(AuthorizationInterceptor(), at: 0)
+        return interceptors
+    }
+}
+
+class AuthorizationInterceptor: ApolloInterceptor {
+    var id: String = ""
+
+    func interceptAsync<Operation: GraphQLOperation>(
+        chain: RequestChain,
+        request: HTTPRequest<Operation>,
+        response: HTTPResponse<Operation>?,
+        completion: @escaping (Swift.Result<GraphQLResult<Operation.Data>, Error>) -> Void) {
+            let appAccountsManager = AppAccountsManager()
+            guard let focusedUserSession = appAccountsManager.focusedAccount else { return }
+            request.addHeader(name: "Authorization", value: "bearer \(focusedUserSession.token)")
+            chain.proceedAsync(request: request,
+                               response: response,
+                               interceptor: self,
+                               completion: completion)
+        }
+}
+
+public struct ApolloError: LocalizedError {
```

Comment:
Question: What is the Error model when a request fails? Since we cannot use ApolloError within Middleware, can we just use enum for Error model